docs/DeveloperGuide.md

```diff
@@ -1,38 +1,564 @@
+
+Step 3: If all arguments are valid, then it finds the transaction in the transactionList by transactionId, which is unique:
+
+`transactionList.getTransactionById(transactionId)`
+
+Step 4: After finding the transaction, a message will be displayed to the user via `Ui.viewTransactionMessage()` method which show the specified transaction's details.
+
+The following sequence diagram shows how the view transaction operation works:
+![viewTransactionSequence](images/ViewTransactionSequence.png)
+
+#### 4.3.5. View Transactions By Status
+
+> The viewTransactionsByStatus feature is facilitated by the TransactionList class. It extends the command class to add a command for users to view the history of finished or ongoing transactions for recording purposes. The feature implements the following commands:
+
+- `find-tx /s finished`: Lists down all the transactions that have been completed.
+- `find-tx /s unfinished`: Lists down all the transactions that are currently still ongoing.
+
+Given below is an example usage scenario and how the command mechanism behaves at each step.
+
+Step 1: The user types in the command in the command line. The CommandParser class checks if the command is valid through the createCommand() method, and either sends an exception, or send the input to the ViewTransactionsByStatus command to be processed.
+
+Step 2: The ViewTransactionsByStatus command checks if the delimiter ('s') is present in the user input through the getArgs() method. If not present, an exception will be thrown. The command also checks whether the input's final argument is 'finished' or 'unfinished' through the isValidArgument() method. An exception will also be thrown if the final argument does not match the required words.
+
+Step 3: Assuming the final argument is 'finish', the entire transactionList will be iterated through, and an array will store the transaction if the transaction.isFinished() status is true.
+
+Step 4: Once the transactionList has been completely iterated through, a message will be displayed to the user via Ui.viewcompletedTransactionsMessage() which lists down all the transactions that have their isFinished() status set to true.
+
+The following sequence diagram shows how the viewTransactionsByStatus operation works:
+
+![viewTransactionsByStatus](images/ViewTransactionByStatusSequence.png)
+
+#### 4.3.6. Update an transaction
+
+> This feature allows the user to update a specific item in the item list
+
+Given below is an example usage scenario and how the command mechanism behaves at each step.
+
+Step 1: The user types in the command in the command line. The CommandParser class checks if the command is valid through the createCommand() method, and either throws an exception, or forwards the input to `UpdateTransactionCommand` class to be processed.
```

Comment:
Would it be better to separate out a sample usage scenario and the explanation on how the code works. It might be clearer to split the sample usage into 1 part and explanation together with a sequence diagram in another